lib/inspec/control_eval_context.rb

```diff
@@ -117,10 +117,21 @@ def to_s
         end
 
         define_method :register_control do |control, &block|
-          if @skip_file || !profile_context_owner.profile_supports_os?
+          if @skip_file
             ::Inspec::Rule.set_skip_rule(control, true)
           end
 
+          unless profile_context_owner.profile_supports_platform?
+            platform = inspec.platform
+            msg = "Platform #{platform.name}/#{platform.release} is not supported with profile '#{profile_context_owner.profile_id}'"
+            ::Inspec::Rule.set_skip_rule(control, msg)
+          end
+
+          unless profile_context_owner.profile_supports_inspec_version?
+            msg = "Current InSpec version #{Inspec::VERSION} is not supported with profile '#{profile_context_owner.profile_id}'"
```

Comment:
Same feedback on this one... "Profile blah is not supported on the current InSpec version (#{Inspec::VERSION})."